core/thread.h

```diff
@@ -233,26 +233,36 @@ namespace MR
      * multi-threaded applications, in practice the \ref image_thread_looping
      * and \ref thread_queue APIs provide much more convenient and powerful
      * ways of developing robust and efficient applications.
-     * 
+     *
      * @{ */
 
     /*! the number of cores to use for multi-threading, as specified in the
      * variable NumberOfThreads in the MRtrix configuration file, or set using
      * the -nthreads command-line option */
     size_t number_of_threads ();
 
+    /*! provides information regarding whether the number of threads has been
+     * initialised, set explicitly, or determined implicitly. This may affect
+     * how particular algorithms choose to launch threads depending on the
+     * presence of a user request. It also prevents repeated testing of
+     * config file / environment variables, since 0 was previously used both
```

Comment:
I don't think documenting the fact that the code was previously buggy really belongs in the docs...? I totally agree that was a bug, but it's already clearly described in the git history, which is where it should be, IMO...

Reply:
True enough. Probably trying to prove to myself at the time that I wasn't overlooking anything...

Reply:
:+1: